src/plugins/data/public/search/i_search.ts

```diff
@@ -47,13 +47,17 @@ export interface IResponseTypesMap {
   [key: string]: IKibanaSearchResponse;
 }
 
-export type ISearchGeneric = <T extends TStrategyTypes = typeof DEFAULT_SEARCH_STRATEGY>(
+export type ISearch<T extends TStrategyTypes = typeof DEFAULT_SEARCH_STRATEGY> = (
   request: IRequestTypesMap[T],
-  options?: ISearchOptions,
-  strategy?: T
+  options?: ISearchOptions
 ) => Observable<IResponseTypesMap[T]>;
 
-export type ISearch<T extends TStrategyTypes = typeof DEFAULT_SEARCH_STRATEGY> = (
+// Service API types
+export interface IStrategyOptions extends ISearchOptions {
+  strategy?: TStrategyTypes;
```

Comment:
I like your idea of passing `strategy` through `options` But the interface name confuse me a little 
What if we rename it  `IStrategyOptions` -> `ISearchGenericOptions`